lib/commonio.c

```diff
@@ -388,12 +388,10 @@ int commonio_lock_nowait (struct commonio_db *db, bool log)
 	lock_file_len = strlen(db->filename) + 6; /* sizeof ".lock" */
 	file = (char*)malloc(file_len);
 	if(file == NULL) {
-		err = ENOMEM;
```

Comment:
Why are you changing this to return 0?

Reply:
The caller (commonio_lock) expects zero return as error and non-zero as success. So this is a bug fix.